### DISCUSSION TITLE
Upgrade Pydantic requirements

### DISCUSSION THREAD

**Discussion by junoriosity** [fastapi/sqlmodel#618] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
pydantic[email]==2.0.2
```

### Description

In the requirements file for my tool I want to make use of the most recent Pydantic version, yet there is a version conflict with sqlmodel

```
sqlmodel 0.0.8 depends on pydantic<2.0.0 and >=1.8.2
```

Can we perhaps release a new version of sqlmodel, such that these issues are resolved?

### Operating System

macOS

### SQLModel Version

0.0.8

### Python Version

3.10

**Comment by Corfucinas** [comment]
Waiting on this as well

**Comment by FilipeMarch** [comment]
This library is abandoned since August 2022 and no signs of recovery

**Reply by antont** [reply]
Not quite so - Tiangolo has worked a lot during the past weeks to get FastAPI over to Pydantic v2. That work is in pretty good shape, and he said 5d ago that will update SQLModel next. https://github.com/tiangolo/fastapi/discussions/9709#discussioncomment-6387149

**Reply by junoriosity** [reply]
@antont That would be terrific news. Many thanks for your input. 🙂

**Reply by Corfucinas** [reply]
> @antont That would be terrific news. Many thanks for your input. 🙂

Sweet, I was afraid I had to actually read Sqlalchemy's documentation

**Reply by FilipeMarch** [reply]
That's awesome news, @antont . Thanks for reporting, I was already getting hopeless.

**Reply by junoriosity** [reply]
@antont Do you think it will happen in the next couple of days or should we expect longer time frames?

**Reply by pintaf** [reply]
Yes, I guess with pydanticv2 and sqlalchemy2 there will be some work to do under the hood of  sqlmodel

**Reply by antont** [reply]
for sqlalchemy2 there's a PR here that some people have already been using in their projects. others have had trouble with forward references there though. https://github.com/tiangolo/sqlmodel/pull/563

**Comment by antont** [comment]
> @antont Do you think it will happen in the next couple of days or should we expect longer time frames?

No idea, I just saw tiangolo mention that in the pydantic v2 fastapi discussion back then. I guess it depends on how busy he is with finishing things there, there's quite a lot of small things apparently.
